src/Webform.js

```diff
@@ -1431,10 +1431,14 @@ export default class Webform extends NestedDataComponent {
         this.checkData(value.data, flags);
         const shouldValidate =
             !flags.noValidate ||
-            flags.fromIFrame ||
+            flags.fromIframe ||
             (flags.fromSubmission && this.rootPristine && this.pristine && flags.changed);
         const errors = shouldValidate
-            ? this.validate(value.data, { ...flags, process: "change" })
+            ? this.validate(value.data, { 
```

Comment:
I'm not sure I understand this change. Rather than an ad-hoc reversal of the `noValidate` flag, doesn't this indicate that actually `noValidate` is being set incorrectly upstream?

Reply:
The answer is yes and no at the same time. Our pdf forms work a bit different than simple forms because they always need to be in contact with iFrame. On every change inside the iFrame the PDF form calls setValue that triggers all other processes like change, recalculations, conditions, validations and so on. 
So, when we want to submit the PDF, the PDF form sends the message to iFrame, and the iFrame sends back the submission object. Before to send this submission to the server, the PDF form needs to validate it. First, the PDF form calls the setValue and pass this submission object as an argument there. The setValue iterates through forms components and call setNestedValue for each of them. If there is no value for some component, it sets the default value together with noValidate flag:
![image](https://github.com/user-attachments/assets/12e67389-2fd3-4a53-9ac7-3008356ee5c5)
In my test form, there was no value for number component and this is the place where noValidate turned to true. 
So, to handle it, we need to set noValidate to false here OR to ignore the noValidate flag in the place where the validation MUST be performed. I chose the second option as found it a bit more secure.
Alternative solution for it can be the passing a dirty flag to the setValue for the value that comes from iFrame, like this: 
![image](https://github.com/user-attachments/assets/e7d443e7-6e3d-475e-b490-a5a1b436d18b)

Reply:
I guess a follow up question here would be why would there ever be a case that we would call `validate()` with the `noValidate` flag being set to `true`? Couldn't we just always set it to `false`, e.g.
```
        const shouldValidate =
            !flags.noValidate ||
            flags.fromIframe ||
            (flags.fromSubmission && this.rootPristine && this.pristine && flags.changed);
        const errors = shouldValidate
            ? this.validate(value.data, {
                ...flags,
                noValidate: false,
                process: 'change'
            })
            : [];
        value.isValid = errors.length === 0;
```